attempt to add interpolate parameter

japhir opened the pull request:
ok having a go at #12.

Don't merge this yet, it seems I don't understand what's happening under the hood with the `gTree` part at all, I just chucked the parameter in there where I thought it's needed.

```
dat <- tibble::tibble(a = 1:10, b = 11:20, c = stats::rnorm(10), d = sample(letters[1:3], 10, TRUE))
ggplot(dat,aes(xmin=a,xmax=b,ymin=0,ymax=1)) + geom_textured_rect(image="~/test.png")
#> Error in gTree(img_grob = img_grob, img_width = img_width, img_height = img_height,  : 
  object 'interpolate' not found
```

japhir committed:
attempt to add interpolate parameter

japhir commented:
Also, it appears to inherit the `interpolate` parameter documentation from geom_raster, which has a different default for interpolation: `#' @inheritParams ggplot2::geom_raster`

clauswilke commented:
You need to feed the interpolate parameter into `rasterGrob` here:
https://github.com/clauswilke/ggtextures/blob/186e40100908dbfa82cd1a09744e7318fea86598/R/texture-grob.R#L83

It doesn't have to be in the `gTree`.

clauswilke commented:
If you add explicit documentation for a parameter it overrides the inherited documentation, I believe.

japhir committed:
move interpolation to rasterGrob

japhir commented:
added the parameter override and moved it to the correct position, but I'm still getting:

```
exiting from: geom_textured_rect(img_width = unit(1, "in"))
Error in grob(raster = raster, x = x, y = y, width = width, height = height,  : 
  object 'interpolate' not found
```

japhir committed:
formatting

clauswilke commented:
Don't add `interpolate` to `default_aes`, it is not an aesthetic.

japhir committed:
remove interpolate from default_aes

japhir commented:
looks like I still haven't resolved the error, but it's bedtime now...

clauswilke commented:
You don't have `interpolate` in the arguments to `draw_panel()`, so of course it doesn't exist inside the function call. You could fish it out of the panel parameters, but honestly a better approach is to make it an argument of `draw_panel()` and remove from `extra_params`.

japhir committed:
move interpolate from extra params to draw_panel

japhir commented:
I've moved it, and by some mystery it does seem to work now! ~~I'm not sure if I haven't left remnants of unintentional side-effects scattered about though :P~~
``` r
library(ggplot2)
library(magick)
#> Linking to ImageMagick 7.0.10.37
#> Enabled features: cairo, fontconfig, freetype, lcms, pango, rsvg, webp, x11
#> Disabled features: fftw, ghostscript
#> Using 4 threads
library(tibble)
devtools::load_all("~/Downloads/ggtextures")
#> Loading ggtextures

# geom_textured_rect
data <- tibble(
   xmin = c(1, 2.5), ymin = c(1, 1), xmax = c(2, 4), ymax = c(4, 3),
   image = list(
     "https://jeroen.github.io/images/Rlogo.png",
     image_read_svg("https://jeroen.github.io/images/tiger.svg")
    )
 )
ggplot(data, aes(xmin = xmin, xmax = xmax, ymin = ymin, ymax = ymax, image = image)) +
   geom_textured_rect(img_width = unit(30, "in"), interpolate = TRUE)
```

![](https://i.imgur.com/msuiRAg.png)

``` r
ggplot(data, aes(xmin = xmin, xmax = xmax, ymin = ymin, ymax = ymax, image = image)) +
   geom_textured_rect(img_width = unit(30, "in"), interpolate = FALSE)
```

![](https://i.imgur.com/bZ9xkTF.png)

``` r
# geom_textured_col
df <- tibble(
   trt = c("a", "b", "c"),
   outcome = c(2.3, 1.9, 3.2),
   image = c(
     "http://www.hypergridbusiness.com/wp-content/uploads/2012/12/rocks2-256.jpg",
     "http://www.hypergridbusiness.com/wp-content/uploads/2012/12/stone2-256.jpg",
     "http://www.hypergridbusiness.com/wp-content/uploads/2012/12/siding1-256.jpg"
   )
 )

ggplot(df, aes(trt, outcome, image = image)) +
  geom_textured_col(img_width = unit(30, "in"), interpolate = TRUE)
```

![](https://i.imgur.com/vwmOXLa.png)

``` r
ggplot(df, aes(trt, outcome, image = image)) +
  geom_textured_col(img_width = unit(30, "in"), interpolate = FALSE)
```

![](https://i.imgur.com/JVsMr3q.png)

``` r
# textured bars
images = c(
  compact = "http://www.hypergridbusiness.com/wp-content/uploads/2012/12/rocks2-256.jpg",
  midsize = "http://www.hypergridbusiness.com/wp-content/uploads/2012/12/stone2-256.jpg",
  suv = "http://www.hypergridbusiness.com/wp-content/uploads/2012/12/siding1-256.jpg",
  `2seater` = "http://www.hypergridbusiness.com/wp-content/uploads/2012/12/mulch1-256.jpg",
  minivan = "http://www.hypergridbusiness.com/wp-content/uploads/2012/12/rocks1-256.jpg",
  pickup = "http://www.hypergridbusiness.com/wp-content/uploads/2012/12/wood3-256.jpg",
  subcompact = "http://www.hypergridbusiness.com/wp-content/uploads/2012/12/concrete1-256.jpg"
)

ggplot(mpg, aes(class, image = class)) +
  geom_textured_bar(img_width = unit(30, "in"), interpolate = TRUE) +
  scale_image_manual(values = images)
```

![](https://i.imgur.com/Fty8W2E.png)

``` r
ggplot(mpg, aes(class, image = class)) +
  geom_textured_bar(img_width = unit(30, "in"), interpolate = FALSE) +
  scale_image_manual(values = images)
```

![](https://i.imgur.com/bJlAid6.png)

``` r
ggplot(mpg, aes(factor(trans), image = class)) +
  geom_textured_bar(img_width = unit(30, "in"), interpolate = TRUE) +
  scale_image_manual(values = images)
```

![](https://i.imgur.com/6sY67d1.png)

``` r
ggplot(mpg, aes(factor(trans), image = class)) +
  geom_textured_bar(img_width = unit(30, "in"), interpolate = FALSE) +
  scale_image_manual(values = images)
```

![](https://i.imgur.com/X3Bh4Nl.png)

<sup>Created on 2020-11-10 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

japhir committed:
undo formatting changes to untouched parts of code

japhir committed:
undo automatic roxygen version update

japhir committed:
remove interpolate from extra_params in geom_textured_bar

japhir commented:
while it does appear that the new rogygen made some changes to the docs layout, I think this could be merged now!

clauswilke commented:
Yes, looks like it's working. I'll review one more time carefully later today. I don't understand the commit that undoes the roxygen version update, though. What exactly did you do there? Roxygen always records the version with which the latest documentation build was performed, and it naturally goes up over time as new roxygen versions are released. The version I currently have installed on my machine is 7.1.1.

japhir committed:
redo roxygen build

japhir commented:
ah ok, I thought you'd find it annoying if I'm using a different version of roxygen from you, so I tried to undo those changes.

Just ran it again, looks like the only change is that value.

japhir commented:
should I squash the commits or will they be shown as a single commit once they're merged anyhow?

clauswilke commented:
They will be squashed on merge.

clauswilke commented:
Thanks!
